cmd/keymasterd/main.go

```diff
@@ -354,9 +412,34 @@ func (state *RuntimeState) checkAuth(w http.ResponseWriter, r *http.Request) (st
 		return "", AuthTypeNone, err
 
 	}
+	if (info.AuthType & requiredAuthType) == 0 {
+		state.writeFailureResponse(w, r, http.StatusUnauthorized, "")
+		err := errors.New("Insufficeint Auth Level")
+		return "", info.AuthType, err
+	}
 	return info.Username, info.AuthType, nil
 }
 
+func (state *RuntimeState) getRequiredWebUIAuthLevel() int {
+	AuthLevel := 0
+	for _, webUIPref := range state.Config.Base.AllowedAuthBackendsForWebUI {
+		if webUIPref == proto.AuthTypePassword {
+			AuthLevel = AuthLevel | AuthTypePassword
```

Comment:
FYI: you can do:
`    AuthLevel |= AuthTypePassword`

Reply:
and will change all of them